accounts/queries.py

```diff
@@ -0,0 +1,277 @@
+        A dict representing the child.
+    """
+    expanded_child = _to_dict(child_object)
+
+    # 1) Change birthday to age in days.
+    age_delta = date.today() - expanded_child.pop("birthday")
+    expanded_child["age_in_days"] = age_delta.days
+
+    # 2) Expand existing conditions in-place.
+    expanded_conditions = dict(expanded_child.pop("existing_conditions").items())
+    expanded_child.update(expanded_conditions)
+
+    # 3) Expand languages in place.
+    expanded_languages = {
+        f"speaks_{langcode}": boolean
+        for langcode, boolean in expanded_child.pop("languages_spoken").items()
+    }
+    expanded_child.update(expanded_languages)
+
+    ga_enum = expanded_child.pop("gestational_age_at_birth")
+
+    gestational_age_in_weeks = _gestational_age_enum_value_to_weeks(ga_enum)
+
+    expanded_child["gestational_age_in_weeks"] = gestational_age_in_weeks
+
+    return expanded_child
+
+
+def _to_dict(model_instance):
+    """Better version of django.forms.models.model_to_dict.
+
+    Args:
+        model_instance: A django model instance.
+
+    Returns:
+        A dictionary formed from a model instance.
+    """
+    opts = model_instance._meta
+    data = {}
+    for f in chain(opts.concrete_fields, opts.private_fields):
+        data[f.name] = f.value_from_object(model_instance)
+    return data
+
+
+def _gestational_age_enum_value_to_weeks(enum_value: int):
+    """Convert enum value on child object to actual # of weeks.
+
+    This enables us to directly query the expanded child object with a
+    scalar value. 0 == "under 24 weeks"; 17 = "Over 40 weeks". To see
+    enumerated values, please reference studies/fields.py.
+    """
+    return min(max(23, enum_value + 23), 40) if enum_value else None
+
+
+@v_args(inline=True)
+class FunctionTransformer(Transformer):
+    def bool_expr(self, bool_term, *others):
+        or_clauses = " ".join(f"or {other}" for other in others)
+        return f"({bool_term} {or_clauses})"
+
+    def bool_term(self, bool_factor, *others):
+        and_clauses = " ".join(f"and {other}" for other in others)
+        return f"({bool_factor} {and_clauses})"
+
+    def relation_expr(self, name, comparator, other):
+        """Translation rule for relational expressions.
+
+        We have special, hardcoded behavior here for gender testing
+        because we don't want to pollute the grammar. Better to enforce
+        on the transformer level the particulars of our application.
+
+        Args:
+            name: string of format CNAME from common.lark
+            comparator: one of =, <, <= >, >=, !=
+            other: string of format CNAME from common.lark
+
+        Returns:
+            The string form of the intended python expression.
+
+        Raises:
+            GrammarError if a gender test does not obey the contract.
+        """
+        if (
+            name == "gender"
+            and comparator not in ("=", "!=")
+            and other not in ("f", "m", "o", "na")
```

Comment:
I think this should be `name == "gender" and ( comparator not in ("=", "!=") or other not in ("f", "m", "o", "na"))` - not all of these anded together. Right now `gender = female` and `gender > o` are both fine.

Reply:
gah - totally right. Let's fix that.